tests/test_unit/test_cdf_tk/test_commands/test_auth.py

```diff
@@ -201,3 +203,21 @@ def test_get_capabilities_by_loader_dm_only_project(toolkit_client_approval: App
     cap_types_dm_only = {type(c) for c in caps_dm_only}
     assert AssetsAcl not in cap_types_dm_only
     assert RelationshipsAcl not in cap_types_dm_only
+
+
+def test_simulate_app_config_acl_unhashable_type_error():
+    app_config_acl = AppConfigAcl(
+        actions=[AppConfigAcl.Action.Read, AppConfigAcl.Action.Write], scope=AppConfigScope(apps=["SEARCH"])
+    )
+
+    # Older version of AuthCommand()._merge_capabilities() would raise an unhashable type error
+    with pytest.raises(TypeError) as err_info:
+        # This will trigger the unhashable type error - logic used in older version of AuthCommand()._merge_capabilities()
+        _ = {(type(app_config_acl), app_config_acl.scope): {app_config_acl.actions}}
+
+    assert err_info.value.args[0] == "unhashable type: 'list'"
+
+    # Newer version of AuthCommand()._merge_capabilities() would not raise an unhashable type error
+    merged_capabilities = AuthCommand._merge_capabilities([app_config_acl])
+    assert len(merged_capabilities) == 1
+    assert type(merged_capabilities[0]) is AppConfigAcl
```

Comment:
This is a very ad-hoc test that is tailored for this specific bug. Instead I suggest you make test of the merge functionality with something like this. Note that the auth command is undertested, so it is good to add some appropriate tests to it.

```suggestion
@pytest.mark.parametrize(
    "input_capabilities,expected_merged",[
    pytest.param(
        [
            AppConfigAcl(actions=[AppConfigAcl.Action.Read], scope=AppConfigScope(apps=["SEARCH"])),
            AppConfigAcl(actions=[AppConfigAcl.Action.Write], scope=AppConfigScope(apps=["SEARCH"])),
        ],
        [
            AppConfigAcl(actions=[AppConfigAcl.Action.Read, AppConfigAcl.Action.Write], scope=AppConfigScope(apps=["SEARCH"])),
        ],
        id="Merge AppConfigAcl with same scope",
    )
    ]
)
def test_merge_capabilities(input_capabilities: list[Capability], expected_merged: list[Capability]) -> None:
    merged = AuthCommand._merge_capabilities(input_capabilities)

    assert [item.dump() for item in merged] == [item.dump() for item in expected_merged]

```

Reply:
Got it! from the comment I understood of adding test which will show what the error was and is the fix solving it or not. I'll update the test and make the method public.